realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -118,10 +119,20 @@ public boolean isValid() {
         if (!managedMode) {
             return false;
         }
-
+        if (realm == null || realm.isClosed()) {
```

Comment:
`managedMode` is actually redundant, checking null for `realm` is enough. But i am fine to solve it in another PR.